lib/galaxy/managers/jobs.py

```diff
@@ -109,7 +109,9 @@ def replace_dataset_ids(path, key, value):
             conditions.append(
                 model.Job.state.in_([model.Job.states.NEW,
                                      model.Job.states.QUEUED,
+                                     model.Job.states.NOT_READY,
```

Comment:
I think
```
~model.Jobs.state.in_([model.Job.state.ERROR, model.Job.state.FAILED])
```
should work and fix the API test.

Reply:
I've opened a PR against your branch in https://github.com/natefoo/galaxy/pull/11 that implements this. That should make the test more robust.

Reply:
Do we also need `model.Job.states.DELETED` if we are looking to exclude terminal jobs? We could also just use `model.Job.terminal_states` which has the value `[states.OK, states.ERROR, states.DELETED]` or `model.Job.non_ready_states` which includes all the others (except `model.Job.states.DELETED_NEW`, which maybe it should).

In fact, what about:

```python
class Job(...):
    states = Bunch(...)  # same as it is now
    terminal_states = set([states.OK, states.ERROR, states.DELETED, states.FAILED])
    non_terminal_states = set(states.values()).difference(terminal_states)
```

And maybe a few more sets as class attributes describing other conditions such as "pre-running" (everything that used to be queued), "running" (running and later but non-terminal), etc? If everything used these sets instead of building their own we should run in to fewer problems in the future?

Reply:
Didn't see that we also had `model.Job.states.DELETED`, yes, that should also go in there.
terminal or non_terminal would both not be correct, we want everything that hasn't failed yet (or is OK).

Reply:
so something like `potentially_ok_states = non_terminal_states.union({states.OK})`

Reply:
Ok, I'll work on it.